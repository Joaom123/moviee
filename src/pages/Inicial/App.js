import React, {Component} from 'react';
import SearchForm from "../../components/SearchForm";
import Card from "../../components/Card";
import "../../style.css";
import "materialize-css/dist/css/materialize.min.css";
import "materialize-css/dist/js/materialize.min";
import apiService from "../../services/apiService";
import Button from "../../components/Button";

class App extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchValue: "marvel",
            page: 1,
            filmes:
                [
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    },
                    {
                        "Title": "Captain Marvel",
                        "Year": "2019",
                        "imdbID": "tt4154664",
                        "Type": "movie",
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
                    }
                ]
        };

        this.searchMovies = this.searchMovies.bind(this);
        this.onButtonClick = this.onButtonClick.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    searchMovies (searchValue) {
        apiService.ListaFilmes({searchValue})
            .then(response => {
                if(response.data.Response === 'True'){
                    this.setState({
                        filmes: response.data.Search
                    })
                }
            })
            .catch(error => {
                console.log(error.response)
            });
    }

    onButtonClick () {
        let page = this.state.page + 1;
        let searchValue = this.state.searchValue;

        apiService.ListaFilmes({searchValue, page})
        .then(response => {
            if(response.data.Response === 'True'){
                console.log(response);
                this.setState({
                    page: page,
                    filmes: response.data.Search
                });
            }
        })
    }

    handleChange (event) {
        this.setState({
            searchValue: event.target.value
        });
    }

    render() {
        return (
            <main>
                <div className="container">
                    <SearchForm
                        onSubmit={this.searchMovies}
                        searchValue={this.state.searchValue}
                        handleChange={this.handleChange}
                    />
                    <section className="grid-template">
                        {
                            this.state.filmes.map( filme =>
                                <Card filme = {filme}/>
                            )
                        }
                    </section>
                    <Button onClick={this.onButtonClick}/>
                </div>
            </main>
        );
    }
}

export default App;
