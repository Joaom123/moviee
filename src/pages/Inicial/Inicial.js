import React, {Component} from 'react';
import SearchForm from "../../components/SearchForm";
import Card from "../../components/Card";
import "../../style.css";
import "materialize-css/dist/css/materialize.min.css";
import "materialize-css/dist/js/materialize.min";
import apiService from "../../services/apiService";
import Button from "../../components/Button";
import validacaoInput from "../../services/validacaoInputService";

class Inicial extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchValue: "",
            page: 1,
            filmes: [],
        };

        this.onSubmit = this.onSubmit.bind(this);
        this.onButtonClick = this.onButtonClick.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.getMoviesBySearchValueAndPage = this.getMoviesBySearchValueAndPage.bind(this);
    }

    getMoviesBySearchValueAndPage ({searchValue, page = 1}) {
        apiService.ListaFilmes({searchValue, page})
        .then(response => {
            console.log(response);
            if(response.data.Response === 'True'){
                this.setState({
                    page,
                    filmes: response.data.Search
                });
            }else{
                //TODO: Mensagem de erro
            }
        })
        .catch(error => {
            console.log(error);
        })
    }

    onSubmit (searchValue) {
        this.getMoviesBySearchValueAndPage({searchValue});
    }

    onButtonClick () {
        let page = this.state.page + 1;
        let searchValue = this.state.searchValue;

        this.getMoviesBySearchValueAndPage({searchValue, page});
    }

    handleChange (event) {
        let searchValue = event.target.value;

        validacaoInput.search(event);

        this.setState({
            searchValue
        });
    }

    render() {
        return (
            <main>
                <div className="container">
                    <SearchForm
                        onSubmit={this.onSubmit}
                        searchValue={this.state.searchValue}
                        handleChange={this.handleChange}
                    />
                    <section className="grid-template">
                        {
                            this.state.filmes.map( filme =>
                                <Card filme = {filme}/>
                            )
                        }
                    </section>
                    <Button onClick={this.onButtonClick}/>
                </div>
            </main>
        );
    }
}

export default Inicial;
