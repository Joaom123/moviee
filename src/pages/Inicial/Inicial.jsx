import React, {Component} from 'react';
import SearchForm from "../../components/SearchForm";
import Card from "../../components/Card";
import Button from "../../components/Button";
import "../../style.css";
import "materialize-css/dist/css/materialize.min.css";
import "materialize-css/dist/js/materialize.min";
import apiService from "../../services/apiService";

class Inicial extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchValue: "",
            page: 1,
            filmes: [],
        };

        this.onButtonClick = this.onButtonClick.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.getMoviesBySearchValueAndPage = this.getMoviesBySearchValueAndPage.bind(this);
    }

    getMoviesBySearchValueAndPage({searchValue, page = 1}) {
        apiService.listaFilmes({searchValue, page})
        .then(response => {
            console.log(response);
            if (response.data.Response === 'True') {
                this.setState({
                    page,
                    filmes: response.data.Search
                });
            } else {
                //TODO: Mensagem de erro
            }
        })
        .catch(error => {
            console.log(error);
        })
    }

    onButtonClick() {
        let page = this.state.page + 1;
        let searchValue = this.state.searchValue;

        this.getMoviesBySearchValueAndPage({searchValue, page});
    }

    handleChange(event) {
        let searchValue = event.target.value;

        this.setState({
            searchValue
        });

        if(searchValue.length >= 3)
            this.getMoviesBySearchValueAndPage({searchValue});
    }

    render() {
        return (
            <main>
                <div className="container">
                    <SearchForm
                        searchValue={this.state.searchValue}
                        handleChange={this.handleChange}
                    />
                    <section className="grid-template">{
                        this.state.filmes.map(filme =>
                            <Card filme={filme} key={filme.imdbID}/>
                        )
                    }
                    </section>
                    <Button onClick={this.onButtonClick}/>
                </div>
            </main>
        );
    }
}

export default Inicial;
