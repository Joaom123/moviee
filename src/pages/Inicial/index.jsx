import React, {Component} from 'react';
import {ModalMovie, Movies, Pagination, SearchForm} from "../../components";
import {apiService} from "../../services";

class Inicial extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchValue: "",
            page: 1,
            movies: [],
            movie: [],
            totalResults: 0,
            toggleModal: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.getMoviesBySearchValueAndPage = this.getMoviesBySearchValueAndPage.bind(this);
        this.handleChangeOfPage = this.handleChangeOfPage.bind(this);
        this.onCloseModal = this.onCloseModal.bind(this);
    }

    getMoviesBySearchValueAndPage({searchValue, page = 1}) {
        apiService.getMoviesList({searchValue, page})
        .then(response => {
            if (response.data.Response === 'True') {
                this.setState({
                    page,
                    movies: response.data.Search,
                    totalResults: response.data.totalResults
                });
            } else {
                //TODO: Mensagem de erro
            }
        })
        .catch(error => {
            console.log(error);
        })
    }

    handleChangeOfPage(e, page) {
        e.preventDefault();
        let searchValue = this.state.searchValue;

        if (searchValue === "")
            return ;

        if (page < 1)
            return ;

        this.getMoviesBySearchValueAndPage({searchValue, page});
    }

    handleChange(event) {
        let searchValue = event.target.value;

        if (searchValue === " ")
            return ;

        this.setState({searchValue});

        if(searchValue.length >= 3)
            this.getMoviesBySearchValueAndPage({searchValue: searchValue.trim()});
    }

    onCardClick = (imdbId) =>
        apiService.getMovieByImdbId({imdbId})
        .then(response =>
            this.setState({
                movie: response.data,
                toggleModal: true
            })
        );

    onCloseModal = () => this.setState({toggleModal: false});

    render() {
        const {searchValue, movies, totalResults, page, movie, toggleModal} = this.state;

        return (
            <main className="container">
                <SearchForm
                    searchValue={searchValue}
                    handleChange={this.handleChange}
                />
                <Movies
                    movies={movies}
                    onCardClick={this.onCardClick}
                />
                <Pagination
                    totalResults={totalResults}
                    page={page}
                    handleChangeOfPage={this.handleChangeOfPage}
                />
                <ModalMovie
                    toggleModal={toggleModal}
                    onCloseModal={this.onCloseModal}
                    movie={movie}
                />
            </main>
        );
    }
}

export default Inicial;
