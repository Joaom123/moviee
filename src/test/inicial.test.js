import React from "react";
import Inicial from "../pages/Inicial/Inicial";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup)

it('snapshot pÃ¡gina inicial, sem dados', () => {
    const {asFragment} = render(<Inicial/>);

    expect(asFragment(<Inicial/>)).toMatchSnapshot();
});

//Snapshot com mock

