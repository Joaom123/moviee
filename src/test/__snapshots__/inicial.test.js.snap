// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot página inicial, sem dados 1`] = `
<DocumentFragment>
  <main>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <form
          class="col s12"
        >
          <div
            class="row"
          >
            <div
              class="input-field col s12 m6 offset-m3"
            >
              <input
                class="validate"
                id="nomeFilme"
                name="nomeFilme"
                type="text"
                value=""
              />
              <label
                for="nomeFilme"
              >
                Nome do Filme
              </label>
            </div>
          </div>
        </form>
      </div>
      <section
        class="moviesCards"
      />
      <button>
        Próxima página
      </button>
    </div>
  </main>
</DocumentFragment>
`;
